### PR TITLE
MkDocs Documentation

### PR DESCRIPTION
**Pull request by activivan**
Supersedes #508 and #21

After struggling with signing my old commits, this is the new pull request for the MkDocs Documentation
https://squidfunk.github.io/mkdocs-material/

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: docs 
+on:
+  push:
+    branches:
+      - master
```
Comment (datamweb) [inline .github/workflows/docs.yml:5]:
```suggestion
      - develop
```

**Review by datamweb — CHANGES_REQUESTED** [review]
Wait for @kenjis  to confirm before applying.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (kenjis) [reply .github/workflows/docs.yml:5]: The HTML documentation is the official one. So it is better that using `master` which is the released version.

Comment (datamweb) [reply .github/workflows/docs.yml:5]: thanks!

**Review by datamweb — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -6,3 +6,4 @@ phpunit
 composer.lock
 .DS_Store
 .idea/
+site
```
Comment (kenjis) [inline .gitignore:9]: What is `site`?

**Comment by datamweb** [comment]
Can you fix this issue?

Screenshot 2022-11-08 092647

File: .gitignore (hunk 2 above)
Comment (jozefrebjak) [reply .gitignore:9]: @kenjis The site folder is generated with a build command. I believe the author of a PR builds the site locally, but with CI it is maybe not necessary to include it in gitignore.

Comment (activivan) [reply .gitignore:9]: Should I remove it then?

Comment (kenjis) [reply .gitignore:9]: No problem. Keep it.

**Comment by activivan** [comment]
This seems to have fixed it

**Comment by kenjis** [comment]
```console
$ pip3 install mkdocs
$ pip3 install mkdocs-material
$ mkdocs --version
mkdocs, version 1.4.2 from /usr/local/lib/python3.10/site-packages/mkdocs (Python 3.10)
$ mkdocs build
$ mkdocs serve
```

**Review by kenjis — APPROVED** [review]
LGTM! 👍

**Comment by datamweb** [comment]
@activivan Thanks for this PR.

@lonnieezell Please check here, if possible, create a subdomain.

**Comment by kenjis** [comment]
Something wrong with GitHub Pages:
https://kenjis.github.io/codeigniter-shield/

**Comment by kenjis** [comment]
I pushed `gh-pages` branch.

```console
$ mkdocs gh-deploy --force
INFO     -  Cleaning site directory
INFO     -  Building documentation to directory: /Users/kenji/work/codeigniter/tmp/shield/site
INFO     -  Documentation built in 0.71 seconds
WARNING  -  Version check skipped: No version specified in previous deployment.
INFO     -  Copying '/Users/kenji/work/codeigniter/tmp/shield/site' to 'gh-pages' branch and pushing to GitHub.
Enumerating objects: 81, done.
Counting objects: 100% (81/81), done.
Delta compression using up to 4 threads
Compressing objects: 100% (65/65), done.
Writing objects: 100% (81/81), 532.13 KiB | 2.60 MiB/s, done.
Total 81 (delta 15), reused 3 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (15/15), done.
remote: 
remote: Create a pull request for 'gh-pages' on GitHub by visiting:
remote:      https://github.com/codeigniter4/shield/pull/new/gh-pages
remote: 
To github.com:codeigniter4/shield.git
 * [new branch]      gh-pages -> gh-pages
INFO     -  Your documentation should shortly be available at: https://codeigniter4.github.io/shield/
```

**Comment by kenjis** [comment]
I set the GitHub Pages branch.
Okay, no problem.
https://codeigniter4.github.io/shield/

**Pull request merged**